File: BackEnd/Models/Passagem.cs
using System.ComponentModel.DataAnnotations;

public class Passagem
{
    [Key]
    public int Id { get; set; }
    public string? Classe { get; set; }
    public decimal Preco { get; set; }
    public Pagamento? Pagamento { get; set; }
    public int PagamentoId { get; set; }
    public Portao? Portao { get; set; }
    public int PortaoId { get; set; }
    public Voo? Voo { get; set; }
    public int VooId { get; set; }
    public Passageiro? Passageiro { get; set; }
    public int PassageiroId { get; set; }
}

File: FrontEnd/AppPassagem/src/app/models/Passagem.ts
import { Pagamento } from "./Pagamento";
import { Passageiro } from "./Passageiro";
import { Portao } from "./Portao";
import { Voo } from "./Voo";

export class Passagem {
    id: number = 0;
    classe: string = "";
    preco: number = 0;
    pagamento: Pagamento;
    pagamentoId: number = 0;
    portao: Portao;
    portaoId: number = 0;
    voo: Voo;
    vooId: number = 0;
    passageiro : Passageiro;
    passageiroId: number = 0;

    constructor(pagamento: Pagamento, portao : Portao, voo: Voo, passageiro : Passageiro){
        this.pagamento = pagamento;
        this.portao = portao;
        this.voo = voo;
        this.passageiro = passageiro;
    }
}